source/wp-content/plugins/wporg-internal-notes/src/store/reducer.js

```diff
@@ -44,14 +54,28 @@ export const reducer = ( state = DEFAULT_STATE, { note, notes, noteId, totalNote
 				totalNotes,
 				notes,
 			};
+		case PREPEND_NOTES:
+			const prependedNoteIds = notes.map( ( prependedNote ) => prependedNote.id );
+
+			return {
+				...state,
+				totalNotes: state.totalNotes + notes.length,
+				notes: [ ...notes, ...state.notes ],
+				isCreated: [ ...state.isCreated, ...prependedNoteIds ],
+			};
 		case APPEND_NOTES:
-			const noteIds = notes.map( ( appendedNote ) => appendedNote.id );
+			const appendedNoteIds = notes.map( ( appendedNote ) => appendedNote.id );
 
 			return {
 				...state,
-				totalNotes,
+				totalNotes: state.totalNotes, // Just retrieving more notes, not changing the total.
```

Comment:
I understand why this is the case, but this still seems strange. For consistency I would use still use `totalNotes` from the action here, even though you know it's not changed. Does that have unexpected effects?

Reply:
The `totalNotes` value is ultimately [derived from](https://github.com/WordPress/wporg-internal-notes/blob/47eddbee8bf2a6e2790bcdc4f22eb08e84ca71cd/source/wp-content/plugins/wporg-internal-notes/includes/class-rest-controller.php#L193) `WP_Query->found_posts`. So in the case of appending more notes, the query is using an offset, so the `found_posts` value won't represent _all_ of the note posts, only those selected with the offset.

Reply:
Ah, I see - it's the total _for that query_. Maybe you shouldn't pass through totalNotes in the append/prepend actions to make it more clear that those are not used?

Reply:
Fair enough. That also simplifies things a little bit: 58a8a48